tests/test_data_helpers.py

```diff
@@ -122,14 +147,24 @@ def test_strip_html_tags(self):
 
     def test_format_from_obj(self):
         """Test that format strings can be passed to objects as expected."""
-        date = self.obj_from_fixture("date_expression.json")
-        formatted = data_helpers.format_from_obj(
-            date, "{begin} - {end} ({expression})")
-        self.assertEqual(formatted, "1905 - 1980 (1905-1980)")
-        with self.assertRaises(KeyError) as excpt:
-            formatted = data_helpers.format_from_obj(
-                date, "{start} - {end} ({expression})")
-        self.assertIn("was not found in this object", str(excpt.exception))
+        for fixture, format_string in [
+            ("date_expression.json", "{begin} - {end} ({expression})"),
+            ("date_expression.json", None),
+        ]:
+            date = self.obj_from_fixture(fixture)
+            if format_string is None:
+                self.assertRaises(
+                    Exception, data_helpers.format_from_obj, date, None)
```

Comment:
```suggestion
                self.assertRaises(
                    Exception, data_helpers.format_from_obj, date, None)
```
I'd be interested in thoughts on this, but I've always seen this done like:

```
with self.assertRaises(Exception):
    data_helpers.format_from_obj(date, format_string)
```
Not sure what best practice is though...

Reply:
So, I don't really know best practice either.

From reading [the documentation](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises), it feels like you can do either, but I'm not sure what the advantage of writing the code inline rather than a function is. Given that we were passing args, I thought the function method was the preferred way due to documentation.

Reply:
From my understanding it seems like you'd want to use the `with` block if you wanted to assign that exception to a variable so you can inspect it (see line 162). Given that this is raising a very generic `Exception` I think it might be wise to use a similar pattern to the below to check the exception message? Thoughts?

Reply:
Sure, that makes sense to me. I'll make the changes.